error: [Prusti: invalid specification] accessing fields of #[trusted] types is not allowed
  --> $DIR/trusted.rs:21:14
   |
21 |     let _b = a.value;   //~ ERROR: Accessing fields of a trusted type.
   |              ^^^^^^^
   |
   = help: you might want to mark the function as #[trusted]
note: the type of this place is marked as #[trusted]
  --> $DIR/trusted.rs:20:13
   |
20 |     let a = TrustedBox::new(1);
   |             ^^^^^^^^^^^^^^^^^^

error: [Prusti: invalid specification] accessing fields of #[trusted] types is not allowed
  --> $DIR/trusted.rs:40:14
   |
40 |     let _b = a.value;   //~ ERROR: Accessing fields of a trusted type.
   |              ^^^^^^^
   |
   = help: you might want to mark the function as #[trusted]
note: the type of this place is marked as #[trusted]
  --> $DIR/trusted.rs:39:13
   |
39 |     let a = GenericTrustedBox::new(1);
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

